examples/platforms/nrf528xx/src/radio.c

```diff
@@ -444,13 +453,17 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
     bool result;
 
+    sCurrentChannel = aChannel;
     nrf_802154_channel_set(aChannel);
     if (nrf_802154_state_get() == NRF_802154_STATE_SLEEP)
     {
         // Enable FEM before RADIO leaving SLEEP state.
         nrf5FemEnable();
     }
 
+    sDefaultTxPower = sDefaultTxPower < sMaxTxPowerTable[sCurrentChannel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN]
+                          ? sDefaultTxPower
+                          : sMaxTxPowerTable[sCurrentChannel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN];
```

Comment:
This code may cause the same channel uses different output powers.
For example:  We assume that the max power is 20dBm for the channel 11. The max power is 18dBm for the channel 25. The current channel `sCurrentChannel ` is channel 11, the current `sDefaultTxPower ` is 20dBm.

<1> Set the channel to 25, then `sDefaultTxPower ` will be set to 18 dBm.
<2> Set the channel to 11, then `sDefaultTxPower ` will still be set to 18 dBm.

Reply:
Echo. I don't quite get what is the purpose here. Why not get rid of `sDefaultTxPower` and use local variable?